### PR TITLE
Add deposit, withdraw, redeem tokenized vault hooks

### PR DESCRIPTION
**Pull request by oskarvu**
Closes FRO-712

In addition deletes unused development flags.

### CODE REVIEW

File: packages/app/src/domain/tokenized-vault-operations/useVaultDeposit.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+  return useWrite(
+    {
+      ...config,
+      enabled: enabled && assets > 0n && !!receiver,
+    },
+    {
+      onTransactionSettled: async () => {
+        void client.invalidateQueries({
+          queryKey: aaveDataLayer({ wagmiConfig, chainId, account: receiver }).queryKey,
```
Comment (yivlad) [inline packages/app/src/domain/tokenized-vault-operations/useVaultDeposit.ts:49]: Why all of the hooks need to invalidate `aaveDataLayer`? I don't think they change anything in it?

Comment (oskarvu) [reply packages/app/src/domain/tokenized-vault-operations/useVaultDeposit.ts:49]: I thought that maybe changing wallet balance will impact market info, but after second look it won't. I will remove this invalidation.

**Review by yivlad — APPROVED** [review]

File: packages/app/src/domain/tokenized-vault-operations/useVaultWithdraw.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import { BaseUnitNumber } from '../types/NumericValues'
+import { balances } from '../wallet/balances'
+
+interface UseVaultWithdrawArgs {
+  assets: BaseUnitNumber
+  onTransactionSettled?: () => void
+  enabled?: boolean
+}
+
+export function useVaultWithdraw({
```
Comment (krzkaczor) [inline packages/app/src/domain/tokenized-vault-operations/useVaultWithdraw.ts:16]: can we leave a 1 line comment here explaining the difference between this and `useVaultRedeem`?

File: packages/app/wagmi.config.ts [hunk 3]
```diff
@@ -108,12 +109,6 @@ export default defineConfig({
             [gnosis.id]: '0xD499b51fcFc66bd31248ef4b28d656d67E591A94',
           },
         },
-        {
```
Comment (krzkaczor) [inline packages/app/wagmi.config.ts:111]: I am not  sure if i like this. SavingsXDAI is quite different from sDAI on mainnet. Sure, it supports ERC-4626 but that's about it.

File: packages/app/src/domain/tokenized-vault-operations/useVaultDeposit.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  onTransactionSettled,
+  enabled = true,
+}: UseVaultDepositArgs): ReturnType<typeof useWrite> {
+  const client = useQueryClient()
+  const wagmiConfig = useConfig()
+  const chainId = useChainId()
+
+  const { address: receiver } = useAccount()
+  const assets = toBigInt(_assets)
+  const vault = useContractAddress(savingsDaiConfig.address)
```
Comment (krzkaczor) [inline packages/app/src/domain/tokenized-vault-operations/useVaultDeposit.ts:31]: vault should be an arg to this function. Then also we could easily drop `savingsDaiConfig` rename.

File: packages/app/.env.development [hunk 5]
```diff
@@ -7,12 +7,8 @@ VITE_FEATURE_TOS_REQUIRED=0
 VITE_FEATURE_TOP_BANNER=0
 VITE_FEATURE_CONFIGURABLE_SLIPPAGE=1
 
-# development
-VITE_DEV_AIRDROP_BADGE=1
```
Comment (krzkaczor) [inline packages/app/.env.development:11]: 👌

File: packages/app/src/domain/tokenized-vault-operations/useVaultDeposit.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+import { useContractAddress } from '../hooks/useContractAddress'
+import { ensureConfigTypes, useWrite } from '../hooks/useWrite'
+import { allowance } from '../market-operations/allowance/query'
+import { CheckedAddress } from '../types/CheckedAddress'
+import { BaseUnitNumber } from '../types/NumericValues'
+import { balances } from '../wallet/balances'
+
+export interface UseVaultDepositArgs {
+  token: CheckedAddress
+  assets: BaseUnitNumber
```
Comment (krzkaczor) [inline packages/app/src/domain/tokenized-vault-operations/useVaultDeposit.ts:14]: I don't particulary like naming scheme here.  IMO:
* vault (this is missing)
* token -> assetToken
* assets -> assetsValue

WDYT?

File: packages/app/src/domain/tokenized-vault-operations/useVaultDeposit.ts [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  vault: CheckedAddress
+  assetToken: CheckedAddress
+  assetsAmount: BaseUnitNumber
+  onTransactionSettled?: () => void
+  enabled?: boolean
+}
+
+export function useVaultDeposit({
+  vault,
+  assetToken,
```
Comment (krzkaczor) [inline packages/app/src/domain/tokenized-vault-operations/useVaultDeposit.ts:21]: whyt about reading this value from the vault? one less arg to pass = one less possible mistake to make.

**Review by krzkaczor — APPROVED** [review]

**Review by krzkaczor — APPROVED** [review]

**Pull request merged**